File: 프로그래머스/0/181930. 주사위 게임 2/주사위 게임 2.js
function solution(a, b, c) {
    const uniqueCount = new Set([a, b, c]).size;
    let answer = 0;
    
    if (uniqueCount == 1) {
        return 3*a*3*(a**2)*3*(a**3);
    } else if (uniqueCount == 2) {
        return (a + b + c) * (a ** 2 + b ** 2 + c ** 2);
    } else {
        return a + b + c;
    }
    
    return answer;
}

File: 프로그래머스/0/181930. 주사위 게임 2/주사위 게임 2.py
def solution(a, b, c):
    answer = 0
    numCount = len({a, b, c});
    
    baseSum = a + b + c
    squareSum = a**2 + b**2 + c**2
    cubeSum = a**3 + b**3 + c**3
    
    if numCount == 1:
        return baseSum * squareSum * cubeSum;
    elif numCount == 2:
        return baseSum * squareSum;
    else:
         return baseSum;